### ISSUE TITLE
function-signature multiline should respect annotations

### ISSUE THREAD

**Issue by mklnwt** [pinterest/ktlint#2414]
## Expected Behavior
```kotlin
fun foo(@Bar baz: String) {}
```
This should not be wrapped.

## Current Behavior
It is wrapped to
```kotlin
fun foo(
    @Bar baz: String,
) {}
```

## Your Environment
* Version of ktlint used: `1.0.1`
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): `org.jlleitschuh.gradle.ktlint` version `12.0.2`
* Version of Gradle used (if applicable): `8.5`
* Operating System and version: Windows 11
